src/main/java/org/deepsymmetry/beatlink/VirtualCdj.java

```diff
@@ -688,6 +663,10 @@ private boolean selfAssignDeviceNumber() {
      * @throws IllegalStateException if we are not running
      */
     public List<NetworkInterface> getMatchingInterfaces() {
+        if (proxyingForVirtualRekordbox.get()) {
```

Comment:
This should probably be done after the `ensureRunning()` check to avoid race conditions where `start()` has been called, but has not yet had a chance to set `proxyingForVirtualRekordbox`. Or even more fundamentally to address the fact that this could return a value even if `start()` was never called, and it’s supposed to throw an exception in that case.